bot.user.js

```diff
@@ -567,18 +691,37 @@ window.loop = function() {
             window.playDefence("l");
             return;
         }
-
+		
+		//increase dodge radius when using speed
+		window.speedRadius();
+		window.collisionPoints = window.getCollisionPoints();
+		
         // If no enemies or obstacles, go after what you are going after
-        if (!window.checkCollision(window.getX(), window.getY(), window.getSnakeWidth() * window.collisionRadiusMultiplier)) {
+        if (!window.checkCollision(window.getX(), window.getY(), window.getSnakeWidth()*window.collisionRadiusMultiplier)) {
+			window.setAcceleration(0);
             // Sort the food based on their distance relative to player's snake
             window.sortedFood = window.getSortedFood();
             // Current food
+			//i = 0;
             window.currentFood = window.sortedFood[0];
+			
+			//while (!window.goodPath(window.currentFood)){
+			//	i++;
+			//	window.currentFood = window.sortedFood[i];
+			//}
```

Comment:
As far as I can see, the strategy here seems correct. Hope you find the bug.
I see one bug though, the while loop should probably look something like this:
```
for (food in window.sortedFood){
   if (window.goodPath(food)){
      window.currentFood = food;
      break;
   }
}
```